### PR TITLE
Send confirmation SMS on registration

### PR DESCRIPTION
**Pull request by brucebolt**
What
----

Sends a confirmation text message to registrants who have entered a UK mobile telephone number into the "phone number for text messages" field.

Content is currently awaiting departmental sign-off, so this PR cannot yet be merged.

Example of SMS:
kAJUmrd8

How to review
-------------

- Review code changes.
- Set up and run application locally using a Notify API token.  Ensure text message is sent through Notify interface.

Links
-----

Trello card: https://trello.com/c/8L0b1iIg

### CODE REVIEW

File: README.md
Comment (bilbof) [inline README.md]:
```suggestion
`GOVUK_NOTIFY_SMS_TEMPLATE_ID`, which might involve creating a
```

File: app/controllers/coronavirus_form/check_answers_controller.rb
Comment (bilbof) [inline app/controllers/coronavirus_form/check_answers_controller.rb:40]: I wonder if we also want to update the `smoke_tester?` method to check if the mobile number used is one used by the smoke tester?

File: app/controllers/coronavirus_form/check_answers_controller.rb
Comment (bilbof) [inline app/controllers/coronavirus_form/check_answers_controller.rb:92]: We'll need to remove whitespace from the mobile number here, if this isn't done elsewhere.

File: concourse/tasks/deploy-to-govuk-paas.yml
Comment (bilbof) [inline concourse/tasks/deploy-to-govuk-paas.yml:65]: `GOVUK_NOTIFY_TEMPLATE_ID` will still be set on the app, so we might want to unset that at some point once this is deployed.

File: config/locales/en.yml
Comment (bilbof) [inline config/locales/en.yml]: Do we want to limit the length of `first_name` or `last_name`, and also remove any special characters (e.g. `.`). Spammers could use this to send texts to people with e.g. some spam message, perhaps containing a link. This isn't specific to the SMS though..

File: spec/controllers/coronavirus_form/check_answers_controller_spec.rb
Comment (bilbof) [inline spec/controllers/coronavirus_form/check_answers_controller_spec.rb]:
```suggestion
    it "does not send a text message when no phone number for texting is provided" do
```

**Review by bilbof — COMMENTED** [review]
Looks great, just a few comments.

It sounds like there are other things to consider (e.g. text content) too, so will approve when those are done.

File: app/controllers/coronavirus_form/check_answers_controller.rb
Comment (brucebolt) [reply app/controllers/coronavirus_form/check_answers_controller.rb:40]: Good idea.  I've added a condition for this.

File: concourse/tasks/deploy-to-govuk-paas.yml
Comment (brucebolt) [reply concourse/tasks/deploy-to-govuk-paas.yml:65]: Cool, I'll do this manually after deployment of these changes

File: config/locales/en.yml
Comment (brucebolt) [reply config/locales/en.yml]: I'm hesitant to do this, since some names can be rather long and include special characters [1].  We have around 300 characters remaining in the SMS, so I don't think length is an issue.

Maybe we should have another story to perform some validation on these fields when the data is actually entered so anything that looks like a URL is disallowed, thereby preventing phishing attempts using this form?

[1]: https://www.bbc.co.uk/news/world-us-canada-52557291

File: app/controllers/coronavirus_form/check_answers_controller.rb
Comment (brucebolt) [reply app/controllers/coronavirus_form/check_answers_controller.rb:92]: I think we will have done this when the number is stored in the contact details controller, since the number is being saved as `national_number` which will have already been formatted by the TelephoneNumber gem.

**Review by bilbof — APPROVED** [review]

**Comment by brucebolt** [comment]
I've pushed an additional commit that will override the sender from "GOVUK" to the outgoing number for the National Shielding Service call centre.  Also configured this number in the Notify dashboard.

**Comment by brucebolt** [comment]
This has been rebased against master and tested on staging.

I am awaiting confirmation on the sender name before we can merge and deploy to production.

**Comment by brucebolt** [comment]
We've had confirmation the sender should be `GOVUK` (i.e. the default).  I have therefore removed the final commit for changing the sender.

File: config/locales/en.yml
Comment (bilbof) [inline config/locales/en.yml]: Do you know if this'll be one text based on the length? Just had a thought we might want a test for the length of this content, just in case.

**Review by bilbof — APPROVED** [review]
Excellent work 👍

File: config/locales/en.yml
Comment (brucebolt) [reply config/locales/en.yml]: It will cost multiple text messages to send (5x SMSs IIRC), but it will be delivered as a single message to the recipient's device

Comment (bilbof) [reply config/locales/en.yml]: Wow big. That's good to know 👍

**Pull request merged**
